frontend/src/screens/learner/learner-welcome-modal.tsx

```diff
@@ -2,18 +2,17 @@ import React, { useState } from 'react'
 import { Button, Modal, Stack, Text, Title } from '@mantine/core';
 import { colors } from '@theme';
 import { useApi, useUserPreferences } from '@lib';
-import { ParticipantStudy } from '@api';
 import Waves from '@images/waves.svg';
 import { launchStudy } from '@models';
+import { useParticipantStudies } from './studies';
 
-export const LearnerWelcomeModal: FC<{
-    demographicSurvey: ParticipantStudy | null
-}> = ({ demographicSurvey }) => {
+export const LearnerWelcomeModal: FC = () => {
     const [open, setOpen] = useState(true)
     const api = useApi()
-    const { data: preferences, refetch } = useUserPreferences()
+    const { data: preferences, refetch, isLoading } = useUserPreferences()
+    const { demographicSurvey } = useParticipantStudies();
 
-    if (!demographicSurvey || demographicSurvey.completedAt || preferences?.hasViewedWelcomeMessage) {
+    if (isLoading || !demographicSurvey || demographicSurvey.completedAt || preferences?.hasViewedWelcomeMessage) {
```

Comment:
Preventing a quick flashing when returning to studies dashboard from the landing page